File: bluraySearch.py
import requests
import sys, re
from bs4 import BeautifulSoup


pagenum = 1
baseURL = 'https://www.blu-ray.com/search/?quicksearch='
country = '&quicksearch_country=US'
keyword = '&quicksearch_keyword='
# searchTerm = 'sinners'
searchInput = sys.argv[1]
searchTerm = sys.argv[1].replace(" ","+")
searchSection = '&section=bluraymovies'

url = baseURL + str(pagenum) + country + keyword + searchTerm + searchSection
#print(url)
headers = { 'User-Agent':'Mozilla/5.0' }

r = requests.get(url, headers=headers)
soup = BeautifulSoup(r.text, 'html.parser')
searchresults=soup.find_all('div', style="display: inline-block")



movies = {}

for div in searchresults:
    if div.a:
        if div.a.img:
            movies[div.a['data-productid']]={"title": div.a['title'],"categoryID": div.a['data-categoryid'],"parentID": div.a['data-globalparentid'], "gproductID": div.a['data-globalproductid'], "productID": div.a['data-productid'], "productURL": div.a['href'], "productCover": div.a.img['src']}
        else:
            movies[div.a['data-productid']]={"title": div.a['title'],"categoryID": div.a['data-categoryid'],"parentID": div.a['data-globalparentid'], "gproductID": div.a['data-globalproductid'], "productID": div.a['data-productid'], "productURL": div.a['href'], "productCover": ""}



print("Results:")
if not movies:
    print("No results. Try fewer words or check your spelling.")
else:
    for movie in movies:
        if re.search(searchInput, movies[movie]['title'], re.IGNORECASE):
            print("   ID:", movies[movie]['productID'])
            print("Title:", movies[movie]['title'])
            print(" Page:", movies[movie]['productURL'])
            print("Cover:", movies[movie]['productCover'])
            print("")

File: bluraySearch.php
<?php
    
$searchtitle = ""
$url = "https://www.blu-ray.com/search/?quicksearch=1&quicksearch_country=US&quicksearch_keyword=" . $searchtitle . "&section=bluraymovies";

$options = [
    "http" => [
        "header" => 'User-Agent: Mozilla/5.0'
    ]
];

$context = stream_context_create($options);
$html = file_get_contents($url, false, $context);


//echo $html;

// parse the HTML content using php-html-parser
    $dom = new DOMDocument();
    @$dom->loadHTML($html);
    $xpath = new DOMXPath($dom);
    
    #("//*[contains(@div, 'display: inline-block')]")
    # query = "//div[@style='display: inline-block']";
    #$divs = $xpath->query(//div[@style="display: inline-block"]');
    $atags = $xpath->query("//div[@style='display: inline-block']//a");
    $imgs = $xpath->query("//div[@style='display: inline-block']//a//img");
    $i = 0;
    #echo $ares[0]->getAttribute("title")
//  foreach ($ares as $div) {
//      echo $div->getAttribute("title");
//  }
    
/* 
    [productid] => Array
    (
    [Blu-Ray.com Product ID] => productid
    [Blu-Ray.com Global ID] => globalid
    [Blu-Ray.com URL] => href
    [Title] => Movie Title (YYYY)
    )
    */
foreach ($atags as $atag) {
    $productid = $atag->getAttribute("data-productid");
    $globalid = $atag->getAttribute("data-globalproductid");
    $href = $atag->getAttribute("href");
    $title = $atag->getAttribute("title");
    $img = $xpath->query(".//img", $atag);
    $poster = '';
    if (!empty($img->item(0))) {
        $poster = $img->item(0)->getAttribute("src");
    };
    $movies[$productid] = ["Blu-Ray.com Product ID" => $productid,"Blu-Ray.com Global ID" => $globalid, "Blu-Ray.com URL" => $href, "Title" => $title, "Poster" => $poster];

        }

print_r($movies)
?>
